app/jobs/fastly_log_processor.rb

```diff
@@ -0,0 +1,75 @@
+require 'zlib'
+FastlyLogProcessor = Struct.new(:bucket, :key) do
+  def perform
+    counts = download_counts
+
+    unless Gemcutter::ENABLE_FASTLY_LOG_PROCESSOR
+      # Just log & exit w/out updating stats
+      Delayed::Worker.logger.info "Processed Fastly log counts: #{counts.inspect}"
+      return
+    end
+
+    # Check if this log has already been processed by another job
+    unless Redis.current.setnx(redis_key, 'processing')
+      fail FastlyLogProcessor::AlreadyProcessedError, "Already processed bucket: #{bucket} key: #{key}"
+    end
+
+    # Set a short expiry while updating
+    Redis.current.expire(redis_key, 2.minutes)
+
+    Download.bulk_update(munge_for_bulk_update(counts))
+
+    Redis.current.set(redis_key, 'processed')
+    Redis.current.expire(redis_key, 30.days)
+  end
+
+  # Takes an enumerator of log lines and returns a hash of download counts
+  # E.g.
+  #   {
+  #     'rails-4.0.0' => 25,
+  #     'rails-4.2.0' => 50
+  #   }
+  def download_counts(enumerator = log_lines)
+    enumerator.each_with_object(Hash.new(0)) do |log_line, accum|
+      path, response_code = log_line.split[10, 2]
+      # Only count successful downloads
+      # TODO: should we count 304 not modified responses?
```

Comment:
A 304 is a download attempt, so I think that counts? Right now they would be real downloads so we should probably keep counting them. Any other opinions?